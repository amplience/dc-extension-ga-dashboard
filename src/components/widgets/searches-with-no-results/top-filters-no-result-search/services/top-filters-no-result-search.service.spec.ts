import topFiltersNoResultSearch from './__fixtures__/top-filter-no-results-search';
import getTopFiltersNoResultSearch, {
  getTopFiltersNoResultSearchForCsv,
} from './top-filters-no-result-search.service';
import type { SearchIndex } from 'dc-management-sdk-js';

describe('getTopFiltersNoResultSearch', () => {
  it('should getTopFiltersNoResultSearch return data', async () => {
    const mockedGetItems = jest
      .fn()
      .mockImplementation(() => topFiltersNoResultSearch);
    const mockedTopFiltersNotResultSearchesGet = jest
      .fn()
      .mockImplementation(() => {
        return {
          getItems: mockedGetItems,
        };
      });
    const mockedSearchIndex = {
      related: {
        'top-filters-no-result-search': {
          get: mockedTopFiltersNotResultSearchesGet,
        },
      },
    };
    const params = {
      search: 'q0',
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopFiltersNoResultSearch(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(mockedTopFiltersNotResultSearchesGet).toHaveBeenCalledWith(params);
    expect(mockedGetItems).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
  });
});

describe('getTopFiltersNoResultSearchForCsv', () => {
  it('should getTopFiltersNoResultSearchForCsv return data', async () => {
    const mockedGetItems = jest
      .fn()
      .mockImplementation(() => topFiltersNoResultSearch);
    const mockedTopFiltersNotResultSearchesGet = jest
      .fn()
      .mockImplementation(() => {
        return {
          getItems: mockedGetItems,
        };
      });
    const mockedSearchIndex = {
      related: {
        'top-filters-no-result-search': {
          get: mockedTopFiltersNotResultSearchesGet,
        },
      },
    };
    const params = {
      search: 'q0',
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopFiltersNoResultSearchForCsv(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(mockedTopFiltersNotResultSearchesGet).toHaveBeenCalledWith({
      ...params,
      offset: 0,
      limit: 1000,
    });
    expect(mockedGetItems).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
  });
});
