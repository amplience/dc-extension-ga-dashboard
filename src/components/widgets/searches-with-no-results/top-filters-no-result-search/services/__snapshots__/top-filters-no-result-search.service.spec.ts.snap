// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTopFiltersNoResultSearch should getTopFiltersNoResultSearch return data 1`] = `
Array [
  Object {
    "count": 3,
    "values": Array [
      Object {
        "attribute": "brand",
        "operator": ":",
        "value": "apple",
      },
      Object {
        "attribute": "price",
        "operator": ">",
        "value": "100",
      },
    ],
  },
  Object {
    "count": 1,
    "values": Array [
      Object {
        "attribute": "brand",
        "operator": ":",
        "value": "apple",
      },
    ],
  },
]
`;

exports[`getTopFiltersNoResultSearchForCsv should getTopFiltersNoResultSearchForCsv return data 1`] = `
Array [
  Object {
    "Count": 3,
    "Filter attribute 1": "brand:apple",
    "Filter attribute 2": "price>100",
  },
  Object {
    "Count": 1,
    "Filter attribute 1": "brand:apple",
    "Filter attribute 2": "",
  },
]
`;
